From c9cdd137804e3484c5ba248e70c94e52dd5afa8e Mon Sep 17 00:00:00 2001
From: EasyNet <devel@easynet.dev>
Date: Thu, 7 Sep 2023 10:14:12 +0300
Subject: [PATCH] Fix GRE tests

---
 tests/linux/gre_test | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tests/linux/gre_test b/tests/linux/gre_test
index 688717a..d1c4f7f 100755
--- a/tests/linux/gre_test
+++ b/tests/linux/gre_test
@@ -14,7 +14,7 @@ basic_bringup_body() {
 		IF_GRE_TTL=255
 	atf_check -s exit:0 \
 		-o match:"ip -4 link add tun0" \
-		-o match:"mode gre" \
+		-o match:"type gre" \
 		-o match:"ttl '255'" \
 		-o match:"local '1.2.3.4'" \
 		-o match:"remote '5.6.7.8'" \
@@ -36,7 +36,7 @@ flags_bringup_body() {
 		IF_GRE_TTL=255 IF_GRE_FLAGS="nopmtudisc ignore-df"
 	atf_check -s exit:0 \
 		-o match:"ip -4 link add tun0" \
-		-o match:"mode gre" \
+		-o match:"type gre" \
 		-o match:"ttl '255'" \
 		-o match:"local '1.2.3.4'" \
 		-o match:"remote '5.6.7.8'" \
-- 
2.43.0

